### ISSUE TITLE
Getting Not Found at /oauth url

### ISSUE THREAD

**Issue by krakrjak** [davido/gerrit-oauth-provider#55]
I'm not sure what's going on here, maybe I've upgraded Gerrit too far? I'm currently running 2.12 (I can backtrack if needed) along with the latest binary release of this plugin. After setting it all up and getting keys for both google and github, I am still having issues.

Both google and github are giving errors on the redirect URI even though they match the configuration exactly. I went to the canonical URI for my gerrit installation appending /oath (which is what the Redirect URI is set to for both google and github) and I get a Not Found page. Is this normal behavior?

**Comment by davido** [comment]
What `AUTH` type have you configured in `gerrit.config`? Also check typos. You have written:

 > I went to the canonical URI for my gerrit installation appending /oath

It should be /oauth.

**Comment by krakrjak** [comment]
Yup, I must have become blind to the typo as I put this in both sites (google and github). With OAUTH now working I ran into my next issue, my account was not linked to the existing openid account. Have I buggered that up too?

I have the link-to-existing-openid-accounts = true set for [plugin "gerrit-oauth-provider-google-oauth"] section and I have auth.type = OPENID along with auth.trustedOpenID = ^.*$

What sort of mess have I made for myself?

**Comment by davido** [comment]
What account you have expected to link to? Also note, that it only work in production and not in test, because the domain must exact match.

**Comment by krakrjak** [comment]
Yeah, I think this is due to some changes on my end so the openids will never line up correctly. Thanks for the help. I have a very small user base at the moment so I can carry on, I think.

Do you perhaps have any guidance on merging the usernames or ids? Is it safe to make the new user have the original user's id and delete the original afterwards?

**Comment by davido** [comment]
It's easy to merge users, though. You remove the new user and link new external identity to the existing one. You cannot remove old old user-id.

But, the point is, Google accounts linking should just work, when and only when your domain stays the same.

**Comment by krakrjak** [comment]
That makes total and complete sense. Thank you so much David!

As a note to others, this is the process I went through to fix up my new account. (Note: I'm using H2 so I had to shutdown gerrit to use the gsql command)

select account_id,email_address from account_external_ids where email_address='email@underinspection';

Once I found the appropriate rows from the newly created user along with the original user I ran the following
update account_external_ids set account_id=origuser_id_number where external_id='external_id_to_link';

Notice that account_id is numeric and external_id is a string. I have to do this for each of my users (not too many) because I changed the domain out from underneath them, but this is an easy fix and deleting the aberrant user will be no hassle for me.
